content/usgs-packages/Application_Part4.Rmd

```diff
@@ -0,0 +1,268 @@
+---
+title: "Application - Part 4, publish results"
+date: "9999-04-01"
+author: "Lindsay R. Carr"
+slug: "app-part4"
+image: "img/main/intro-icons-300px/r-logo.png"
+output: USGSmarkdowntemplates::hugoTraining
+parent: Introduction to USGS R Packages
+weight: 2
+draft: true
+---
+
+```{r setup, include=FALSE, warning=FALSE, message=FALSE}
+library(knitr)
+
+knit_hooks$set(plot=function(x, options) {
+  sprintf("<img src='../%s%s-%d.%s'/ title='%s'/>", 
+          options$fig.path, options$label, options$fig.cur, options$fig.ext, options$fig.cap)
+
+})
+
+opts_chunk$set(
+  echo=TRUE,
+  fig.path="static/app-part4/",
+  fig.width = 6,
+  fig.height = 6,
+  fig.cap = "TODO"
+)
+
+knit_hooks$set(addToggle = function(before, options, envir) {
+    if(before) {
+      sprintf('<button class="ToggleButton" onclick="toggle_visibility(\'%1$s\')">Show Answer</button>
+              <div id="%1$s" style="display:none">', opts_current$get('label'))
+    } else {
+      '</div>'
+    }
+})
+
+set.seed(1)
+```
+
+```{r sbtools-auth, include=FALSE}
+# run vizlab::storeSBcreds() once before this can work
+credList <- readRDS(file.path(path.expand('~'), ".vizlab/sbCreds"))
+un <- rawToChar(credList$username)
+pw <- rawToChar(credList$password)
+sbtools::authenticate_sb(un, pw)
+```
+
+In this section, we will complete the workflow and push the finished plots to ScienceBase from within the script. See the [previous section](/usgs-packages/app-part3) to see how we created the plots (or expand the code below).
```

Comment:
still another way to do this would be to have a final Part5 that was just these code sections, with each lesson hidden behind a toggle so you could check it out without peeking at the answers to sections you haven't completed yet. then parts 2-4 could just link to part 5 (and backward) rather than duplicating all this code